### ISSUE TITLE
8$: Improve error on 'await' on class method

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#930]
```javascript
class C {
  constructor() {}
  await myFunction() {
    let myPromise = Promise.resolve(null);
    await myPromise;
  }
}
```
quick-lint-js currently reports:
```
hello.js:3:3: error: unexpected token [E0054]
hello.js:5:5: error: 'await' is only allowed in async functions [E0162]
```
but quick-lint-js should instead treat 'await' as if it was 'async' and report only:
```
hello.js:3:3: error: use 'async' instead of 'await' on class methods [E????]
```

Note: The following example is *legal* because of the line break (`await` is interpreted as a field declaration with no type or initializer) and should produce no diagnostics:
```javascript
class C {
  constructor() {}
  await
  myFunction() {
  }
}
```

Related: https://quick-lint-js.com/errors/E0178/

**Comment by Deep642** [comment]
Hi @strager , I want to contribute in open source, but I am not able to understand what is asked here to be done exactly? can you help me out ?

**Comment by Deep642** [comment]
I want to do this assignment, please assign it to me.

**Comment by strager** [comment]
> I am not able to understand what is asked here to be done exactly?

Create a new diagnostic in quick-lint-js' parser.

Docs: https://quick-lint-js.com/contribute/create-diagnostic/

I think this is the relevant code in the parser, but I didn't double-check: https://github.com/quick-lint/quick-lint-js/blob/8a27f8a473b0db4ff25f196ca0227ff1e5a86285/src/quick-lint-js/fe/parse-class.cpp#L816-L821

**Comment by strager** [comment]
@Deep642 Do you need help with implementing this feature?

**Comment by alyssaperkins** [comment]
@strager Hi there! This is my first attempt at try out an open source contribution.  Can I claim this issue?

**Comment by strager** [comment]
>  Is this a good starter bug

I think yes.

> can I claim it

Sure!

> I may need some help along the way.

I'm in the middle of a cross-country move so I might be unresponsive for a week or two. I'll try my best to answer your questions.

**Comment by alyssaperkins** [comment]
@strager Thanks for getting back with me!  I'll be working on this tonight.  I'll probably have questions soon.

**Comment by alyssaperkins** [comment]
@starger So if I may for clarity, I am writing a new diagnostic that will detect when to throw the error OR I am editing the diagnostic mentioned here? https://github.com/quick-lint/quick-lint-js/blob/8a27f8a473b0db4ff25f196ca0227ff1e5a86285/src/quick-lint-js/fe/parse-class.cpp#L816-L821

**Comment by strager** [comment]
>  I am writing a new diagnostic that will detect when to throw the error OR I am adjusting the diagnostic mentioned here? 

Yes, both. You will need to make a new diagnostic type (in diagnostic-types-2.h) and also code to report the diagnostic (probably in parse-class.cpp). I think that the code you linked is the right code to change for reporting your new diagnostic.

Note that you linked an old version of the code. Here's the latest version for your convenience:

https://github.com/quick-lint/quick-lint-js/blob/4e36b74137ade4dea4480662e30b45349c4daa73/src/quick-lint-js/fe/parse-class.cpp#L985-L990

If you look a few lines up, you'll see a special case for `const`. Maybe we can do something like that but for `await`.

**Comment by alyssaperkins** [comment]
@strager Ok thank you! I think I wrote the diagnostic correctly.  I'm still figuring out how to link things, will be working this weekend or sooner to get that done.  I was using the online chat to get some help, wasn't sure if there is a public discord for help?

**Comment by strager** [comment]
> I was using the online chat to get some help

Yup! I saw your messages, but I wasn't available when you were chatting.

I now have stable internet access, so I am more available.

> wasn't sure if there is a public discord for help?

No. The IRC channel you joined is the right place to discuss quick-lint-js live.

**Comment by alyssaperkins** [comment]
Hey! I am still working on this issue.  Got food poisoning last week so my whole schedule took a hit.  I will be focusing on testing efforts and will get on the IRC if I hit any roadblockers.

**Comment by strager** [comment]
@alyssaperkins Do you need any help?
